package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,352 @@
+            if 'dt' in self._particle_group['data']:
+                ts.data['dt'] = particle_group['data/dt/value'][frame]
+
+        # set frame box dimensions
+        # set triclinic box vectors
+        ts._unitcell[:] = particle_group['box/edges/value'][frame, :]
+
+        # set particle positions
+        frame_positions = particle_group['position/value'][frame, :]
+        n_atoms_now = frame_positions.shape[0]
+        if n_atoms_now != self.n_atoms :
+            raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                             " atoms. MDAnalysis is unable to deal with variable"
+                             " topology!"%(frame, n_atoms_now, self.n_atoms))
+        else :
+            ts.positions = frame_positions
+
+        # set particle velocities
+        if self.has_velocities:
+            frame_velocities = particle_group['velocity/value'][frame, :]
+            n_atoms_now = frame_velocities.shape[0]
+            if n_atoms_now != self.n_atoms :
+                raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                                 " atoms. MDAnalysis is unable to deal with variable"
+                                 " topology!"%(frame, n_atoms_now, self.n_atoms))
+            else :
+                ts.velocities = frame_velocities
+
+        # set particle forces
+        if self.has_forces:
+            frame_forces = particle_group['force/value'][frame, :]
+            n_atoms_now = frame_forces.shape[0]
+            if n_atoms_now != self.n_atoms :
+                raise ValueError("Frame %d has %d atoms but the initial frame has %d"
+                                 " atoms. MDAnalysis is unable to deal with variable"
+                                 " topology!"%(frame, n_atoms_now, self.n_atoms))
+            else :
+                ts.forces = frame_forces
+
+        # unit conversion
+        self._translate_h5md_units()  # fills units dictionary
+        if self.convert_units:
+            #self.ts.time = self.convert_time_to_native(self.ts.time)
+            self.convert_pos_from_native(self.ts.dimensions[:3])
+            self.convert_pos_from_native(self.ts.positions)
+            if self.has_velocities:
+                self.convert_velocities_from_native(self.ts.velocities)
+            if self.has_forces:
+                self.convert_forces_from_native(self.ts.forces)
+
+        return ts
+
+
+    def _read_next_timestep(self) :
+        """read next frame in trajectory"""
+        return self._read_frame(self._frame + 1)
+
+
+    def _translate_h5md_units(self):
+        """converts units from H5MD to MDAnalysis notation
+        and fills units dictionary"""
+
+        velocity_dict = {
+        'nm ps-1': 'nm/ps',
+        'Angstrom ps-1': 'Angstrom/ps'
+        }
+        force_dict = {
+        'kJ mol-1 nm-1': 'kJ/(mol*nm)',
+        'kJ mol-1 Angstrom-1': 'kJ/(mol*Angstrom)'
+        }
```

Comment:
The dicts can be class-level attributes, possibly with underscore and more expressive name such as `_velocity_unit_translation`. The underscore indicates that they are not for users.

Reply:
Or even better: make one big class-level translation dict
```python
#: Translate H5MD units (add link here) to MDAnalysis units.
_unit_translation = {
   'velocity': {
        'nm ps-1': 'nm/ps',
        'Angstrom ps-1': 'Angstrom/ps'
        },
   'force':  {
        'kJ mol-1 nm-1': 'kJ/(mol*nm)',
        'kJ mol-1 Angstrom-1': 'kJ/(mol*Angstrom)'
        }
}
```
that keeps everything tidy.

Reply:
ah ok, so I could just define a `self._unit_translation` prior to the `__init__` like this? That would be very tidy

Reply:
Yes, at the class level, like the `units` attribute itself. It will be available to all instances. It's not meant to be changed by any instance (as it would affect _all_ of them).